scripts/buildorder.py

```diff
@@ -179,36 +187,47 @@ def __init__(self, subpackage_file_path, parent, virtual=False):
             raise Exception("SubPackages should have a parent")
 
         self.name = os.path.basename(subpackage_file_path).split('.subpackage.sh')[0]
-        if "gpkg" in subpackage_file_path.split("/")[-3].split("-") and "glibc" not in self.name.split("-"):
+        if "gpkg" in subpackage_file_path.split("/")[-3].split("-") and not has_prefix_glibc(self.name):
             self.name = add_prefix_glibc_to_pkgname(self.name)
         self.parent = parent
-        self.deps = set([parent.name])
-        self.only_installing = parent.only_installing
+        self.fast_build_mode = parent.fast_build_mode
         self.accept_dep_scr = parent.accept_dep_scr
+        self.depend_on_parent = None
+        self.only_installing = None
         self.excluded_arches = set()
+        self.deps = set()
         if not virtual:
             self.deps |= parse_build_file_dependencies(subpackage_file_path)
+            self.pkg_deps = parse_build_file_dependencies_with_vars(subpackage_file_path, 'TERMUX_SUBPKG_DEPENDS')
             self.excluded_arches |= parse_build_file_excluded_arches(subpackage_file_path)
+            self.depend_on_parent = parse_build_file_variable(subpackage_file_path, "TERMUX_SUBPKG_DEPEND_ON_PARENT")
+            self.only_installing = parse_build_file_variable(subpackage_file_path, "TERMUX_SUBPKG_ONLY_INSTALLING")
+        if not self.depend_on_parent or self.depend_on_parent == "unversioned" or self.depend_on_parent == "true":
+            self.deps |= set([parent.name])
+        elif self.depend_on_parent == "deps":
+            self.deps |= parent.deps
+        self.only_installing = self.only_installing == "true" if self.only_installing else parent.only_installing
         self.dir = parent.dir
 
         self.needed_by = set()  # Populated outside constructor, reverse of deps.
 
     def __repr__(self):
         return "<{} '{}' parent='{}'>".format(self.__class__.__name__, self.name, self.parent)
 
-    def recursive_dependencies(self, pkgs_map, dir_root=None):
+    def recursive_dependencies(self, pkgs_map, dir_root=None, only_installing=False):
         """All the dependencies of the subpackage, both direct and indirect.
         Only relevant when building in fast-build mode"""
         result = []
-        if dir_root == None:
+        if dir_root:
             dir_root = self.dir
-        for dependency_name in sorted(self.deps):
+        for dependency_name in sorted(self.pkg_deps if only_installing else self.deps):
```

Comment:
I actually had the same question above for `self.only_installing`.
What exactly does `{{val}} if {{cond}} else {{other}}` do?

This probably isn't a code problem.
I'm just not familiar with this bit of Python syntax.

Reply:
> I'm just not familiar with this bit of Python syntax.

Here is a link to make this python syntax clearer for you - https://stackoverflow.com/a/11529301

If we consider my example, then it simply selects a list of dependencies depending on what value is set from the variable `self.only_installing`. if it is set to `True`, then the list `self.pkg_dep` will be used. If not (i.e. `False`), then the list `self.deps`.